storage/reader_test.go

```diff
@@ -387,6 +393,8 @@ func TestContentEncodingGzipWithReader(t *testing.T) {
 			gz := gzip.NewWriter(w)
 			gz.Write(original)
 			gz.Close()
+		default:
+			fmt.Fprintf(w, "unrecognized URL %s", r.URL)
```

Comment:
This should be a failure?

Reply:
As discussed offline, the mock just returns this and the test decides how to use it and whether it is a failure state - in this case, the test will fail upon getting such a response.